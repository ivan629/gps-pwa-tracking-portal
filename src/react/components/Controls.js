import React, { useEffect } from 'react';
import Button from '@material-ui/core/Button';
import { useDispatch } from 'react-redux';
import { toggleLocationMonitoring, switchObservedUser } from '../../redux/actions/locationDetectionActions'

import { locationDetectionService } from '../../redux/services/locationDetectionService';

import './Controls.scss'


function Controls({ isLocationMonitoring }) {
    const dispatch = useDispatch();

    useEffect(() => {
            const isMobile = navigator.userAgent.match(/(iPad)|(iPhone)|(iPod)|(android)|(webOS)/i);

            if (isMobile) {
                document.addEventListener('visibilitychange', async () => {
                    if (document.visibilityState === 'hidden') {
                        await dispatch(toggleLocationMonitoring(false));
                    }
                });
            } else {
                window.addEventListener("beforeunload", () => locationDetectionService.stopLocationDetection())
            }
        }
    , []);

    return (
        <div className="controls">
            <Button className="controlButton toggleDetectionButton"
                variant="outlined"
                onClick={() => dispatch(toggleLocationMonitoring(!isLocationMonitoring))}>
                {`${isLocationMonitoring ? 'stop' : 'start'} detection`}
            </Button>
            <Button className="controlButton switchFocusButton" variant="outlined"
                onClick={() => dispatch(switchObservedUser())}>
                switch focus
            </Button>
        </div>
    )
}

export default Controls;
