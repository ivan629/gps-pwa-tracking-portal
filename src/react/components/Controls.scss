.controls {
  height: 50px;
  width: 100%;
  display: flex;

  .controlButton {
    border-radius: 0;
    height: 50px;
    flex: 1;
  }
}
