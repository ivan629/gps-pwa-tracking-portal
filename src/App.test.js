import React from 'react';
import { shallow } from 'enzyme';
import Controls from './react/components/Controls';

describe('Compare correct button text', () => {
  const nextProps = {
    isLocationMonitoring: false
  };

  const wrapper = shallow(<Controls {...nextProps}/>);

  // it('is correct button text', () => {
  //   const text = wrapper.find('.toggleDetectionButton').text();
  //
  //   console.log(text);
  //
  //   expect(text).toEqual(`${nextProps.isLocationMonitoring ? 'stop' : 'start'} detection`);
  // });

  it('renders properly', () => {
    expect(wrapper.debug()).toMatchSnapshot()
  })
});
