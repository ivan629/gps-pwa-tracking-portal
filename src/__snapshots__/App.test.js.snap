// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare correct button text renders properly 1`] = `
"<div className=\\"controls\\">
  <WithStyles(ForwardRef(Button)) className=\\"controlButton toggleDetectionButton\\" variant=\\"outlined\\" onClick={[Function: onClick]}>
    start detection
  </WithStyles(ForwardRef(Button))>
  <WithStyles(ForwardRef(Button)) className=\\"controlButton switchFocusButton\\" variant=\\"outlined\\" onClick={[Function: onClick]}>
    switch focus 1
  </WithStyles(ForwardRef(Button))>
</div>"
`;
